File: go-client-sdk/pkg/models/operations/getdataassessaccountscategories.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetDataAssessAccountsCategoriesSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetDataAssessAccountsCategoriesRequest struct {
	Security GetDataAssessAccountsCategoriesSecurity
}

type GetDataAssessAccountsCategoriesResponse struct {
	CodatAssessDataContractsAccountCategoriesAccountCategories []shared.CodatAssessDataContractsAccountCategoriesAccountCategory
	ContentType                                                string
	StatusCode                                                 int
}

File: typescript-client-sdk/src/sdk/models/operations/getdataassessaccountscategories.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Type } from "class-transformer";


export class GetDataAssessAccountsCategoriesSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetDataAssessAccountsCategoriesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  security: GetDataAssessAccountsCategoriesSecurity;
}

export class GetDataAssessAccountsCategoriesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.CodatAssessDataContractsAccountCategoriesAccountCategory })
  codatAssessDataContractsAccountCategoriesAccountCategories?: shared.CodatAssessDataContractsAccountCategoriesAccountCategory[];

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}